extensions/blocks/contact-form/components/jetpack-newsletter-integration-settings-plugin-state.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import { Button, ExternalLink, Icon } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { installAndActivatePlugin, activatePlugin } from './../../../shared/plugin-management';
+
+const pluginPathWithoutPhp = 'creative-mail-by-constant-contact/creative-mail-plugin';
+const pluginSlug = 'creative-mail-by-constant-contact';
+
+export const pluginStateEnum = Object.freeze( {
+	ACTIVE: 1,
+	INSTALLED: 2,
+	NOT_INSTALLED: 3,
+} );
+
+const CreativeMailPluginIsInstalling = ( { isActivating } ) => {
+	const btnTxt = isActivating ? __( 'Installing…', 'jetpack' ) : __( 'Activating…', 'jetpack' );
+	return (
+		<Button
+			isSecondary
+			icon={ <Icon style={ { animation: 'rotation 2s infinite linear' } } icon="update" /> }
+			disabled
+			aria-label={ btnTxt }
+		>
+			{ btnTxt }
+		</Button>
+	);
+};
+
+const CreativeMailPluginIsNotInstalled = ( {
+	installAndActivateCreativeMailPlugin,
+	isInstalling,
+} ) => {
+	return (
+		<p>
+			<em style={ { color: 'rgba(38, 46, 57, 0.7)' } }>
+				{ __(
+					'To start sending email campaigns, install the Creative Mail plugin for WordPress.',
+					'jetpack'
+				) }
+				<br />
+				{ isInstalling && <CreativeMailPluginIsInstalling /> }
+				{ ! isInstalling && (
+					<Button isSecondary onClick={ installAndActivateCreativeMailPlugin }>
+						{ __( 'Install Creative Mail plugin', 'jetpack' ) }
+					</Button>
+				) }
+			</em>
+		</p>
+	);
+};
+
+const CreativeMailPluginIsInstalled = ( { activateCreativeMailPlugin, isInstalling } ) => {
+	return (
+		<p>
+			<em>
+				{ __(
+					'To start sending email campaigns, activate the Creative Mail plugin for WordPress.',
+					'jetpack'
+				) }
+			</em>
+			<br />
+			{ isInstalling && <CreativeMailPluginIsInstalling isActivating /> }
+			{ ! isInstalling && (
+				<Button isSecondary onClick={ activateCreativeMailPlugin }>
+					{ __( 'Activate Creative Mail Plugin', 'jetpack' ) }
+				</Button>
+			) }
+		</p>
+	);
+};
+
+const CreativeMailPluginIsActive = () => {
+	return (
+		<p>
+			<em>
+				{ __( 'You’re all setup for email marketing with Creative Mail.', 'jetpack' ) }
+				<br />
+				<ExternalLink href="/wp-admin/admin.php?page=creativemail">
+					{ __( 'Open Creative Mail settings', 'jetpack' ) }
+				</ExternalLink>
```

Comment:
There seems to be a minor regression here where subdirectory setups will get the wrong URL - IIRC this was addressed previously by the following snippet:
https://github.com/Automattic/jetpack/pull/16964/files#diff-fa7123df273c65ccfb5a4ca4442206dcL150-L153
```jsx
const getCreativeMailPluginUrl = () => {	
	const adminUrl = get( getJetpackData(), 'adminUrl', false );	
	return `${ adminUrl }admin.php?page=creativemail`;	
};
```